src/comment.rs

```diff
@@ -362,12 +362,18 @@ fn identify_comment(
     };
 
     let (first_group, rest) = orig.split_at(first_group_ending);
+
+    let is_doc_comment_with_our_determination = is_doc_comment || style.is_doc_comment();
     let rewritten_first_group =
         if !config.normalize_comments() && has_bare_lines && style.is_block_comment() {
             trim_left_preserve_layout(first_group, shape.indent, config)?
         } else if !config.normalize_comments()
             && !config.wrap_comments()
-            && !config.format_code_in_doc_comments()
+            && !(
+                // `format_code_in_doc_comments` should only take effect on doc comments,
+                // so we only consider it when this comment block is a doc comment block.
+                is_doc_comment_with_our_determination && config.format_code_in_doc_comments()
+            )
```

Comment:
`is_doc_comment` should always be `false` when we call `identify_comment` from `rewrite_comment`, so there's no need to also check the style.

https://github.com/rust-lang/rustfmt/blob/38659ec6ad5f341cf8eb3139725bf695872c6de7/src/comment.rs#L260-L267

```suggestion
            && !(
                // `format_code_in_doc_comments` should only take effect on doc comments,
                // so we only consider it when this comment block is a doc comment block.
                is_doc_comment && config.format_code_in_doc_comments()
            )
```